naps/cores/video/gamma_corrector.py

```diff
@@ -0,0 +1,36 @@
+from nmigen import *
+
+from naps import ControlSignal, stream_transformer
+from . import ImageStream
+
+__all__ = ["TableGammaCorrector"]
+
+
+class TableGammaCorrector(Elaboratable):
+    """Apply gamma correction to a monochrome image using a pre-computed lookup table"""
+    def __init__(self, input: ImageStream, gamma: float):
+        self.input = input
+        self.bpp = len(input.payload)
+        self.output = ImageStream(self.bpp)
+
+        self.gamma = gamma
+
+    def elaborate(self, platform):
+        m = Module()
+
+        stream_transformer(self.input, self.output, m, latency=1)
+        input_transaction = self.input.ready & self.input.valid
+
+        # compute the gamma lookup table with the formula
+        # out = in ^ gamma, where in and out are 0-1 and ^ is exponentiation
+        max_pix = 2**self.bpp - 1
+        lut = list(int(max_pix*((v/max_pix)**self.gamma)) for v in range(max_pix+1))
```

Comment:
Is there some specific reason for rounding down, instead of rounding the values?

Reply:
No, it's probably better to round normally. I will fix that.